solr/core/src/java/org/apache/solr/cloud/api/collections/CreateCollectionCmd.java

```diff
@@ -560,10 +559,16 @@ public static void createCollectionZkNode(DistribStateManager stateManager, Stri
           }
 
           collectionProps.remove(ZkStateReader.NUM_SHARDS_PROP);  // we don't put numShards in the collections properties
+          collectionProps.remove(ZkStateReader.CONFIGNAME_PROP); // we don't write configName on a zk collection node
 
-          ZkNodeProps zkProps = new ZkNodeProps(collectionProps);
-          stateManager.makePath(collectionPath, Utils.toJSON(zkProps), CreateMode.PERSISTENT, false);
-
+          if (collectionProps.size() > 0) {
+            ZkNodeProps zkProps = new ZkNodeProps(collectionProps);
+            // create a node with collectionProps data
+            stateManager.makePath(collectionPath, Utils.toJSON(zkProps), CreateMode.PERSISTENT, false);
```

Comment:
In some separate issue, if I get around to it, I'll propose dropping these props here since I think nothing reads it.

Reply:
It seems no other collection properties are added into props.

Reply:
Try to create one: https://nightlies.apache.org/solr/draft-guides/solr-reference-guide-main/collection-management.html#create  with `property.foo=bar` and see if foo=bar winds up in that JSON.  I think it will be there.

Reply:
I tried creating a collection with `property.foo`, 
http://localhost:8983/solr/admin/collections?action=CREATE&name=test&property.foo=bar&collection.configName=test&numShards=1
but foo=bar wasn't written on the node as a metadata.

Reply:
Great; lets then not write props since there's nothing there.